components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,254 @@
+import { collection, query, where, getDocs } from "firebase/firestore"
+import ErrorPage from "next/error"
+import { useEffect, useState } from "react"
+import { useMediaQuery } from "usehooks-ts"
+import { useAuth } from "../auth"
+import { Col, Row, Spinner } from "../bootstrap"
+import { usePublicProfile } from "../db"
+import { firestore } from "../firebase"
+import {
+  Header,
+  HeaderTitle,
+  StyledContainer
+} from "./StyledNewsfeedComponents"
+import { AlertCard } from "components/AlertCard/AlertCard"
+
+type NotificationProps = {
+  bodyText: string
+  court: string
+  header: string
+  id: string
+  subheader: string
+  timestamp: string
+  topicName: string
+  type: string
+}
+
+type Notifications = NotificationProps[]
+
+export default function Newsfeed() {
+  const isMobile = useMediaQuery("(max-width: 768px)")
+
+  const { user } = useAuth()
+  const uid = user?.uid
+  const { result: profile, loading } = usePublicProfile(uid)
+  const subscriptionRef = collection(
+    firestore,
+    `/users/${uid}/userNotificationFeed/`
+  )
```

Comment:
I think this needs to be pulled into the `notificationQuery` function, since we don't want to create a subscriptionRef every re-render. This was causing an infinite loop when I was playing around with this branch